### PR TITLE
Use precompiled regular expressions

### PR DESCRIPTION
**Pull request by databus23**
This commit refactors improves the following things.
* Move config related code into seperate file
* Change loadConfig into NewConfig that works on a reader and can be used in tests as well
* Precompile regular expressions during config load
* Reduce signal handler to SIGINT and SIGQUIT
* Reduce netsing in LogEvent function by using a jump label

### CODE REVIEW

File: filter.go [hunk 1]
```diff
@@ -20,69 +19,62 @@ var (
...
 				}
+			} else {
+				value, err = GetValueFromStruct(event, labelSpec)
+			}
 
-				l[labelKey] = value
+			if err != nil {
+				glog.Errorf("Could not get metric label: %v", err)
+				//TODO: Shouldn't this be a continue?
+				break
```
Comment (jknipper) [inline filter.go]: @databus23 Maybe we should skip the whole metric if there is a problem with getting the labels?

Comment (databus23) [reply filter.go]: My point is a `break` in this place will leave the outer for loop, basically exiting from processing any more metrics that come later in the array. A `continue` would skip the metric with missing labels but jump to the top and continue to process other metrics. Correct?

Comment (jknipper) [reply filter.go]: As far as i see `break` will only exit the inner for loop and will continue on the next metric. Problem is that labels are not well defined and depend on involved objects.

Comment (databus23) [reply filter.go]: But the thing is we are already in the outer loop here.

Comment (databus23) [reply filter.go]: a no, ok I see know, there is another for loop. ok then a break will only skip the label. Which is probably not what we want a `continue OUTER` would be better to skip the metric altogether. I will write a test case for that

Comment (jknipper) [reply filter.go]: Aren't we still in `for _, metric := range er.Config.Metrics {...}` after `break`?

Comment (jknipper) [reply filter.go]: Yes, `continue OUTER` makes sense.

**Comment by databus23** [comment]
I added a test to ensure we are skipping metrics with errors during label generation

**Comment by jknipper** [comment]
Very good, going to merge it.

**Pull request merged**